src/server/game/World/IWorld.h

```diff
@@ -512,10 +514,13 @@ enum Rates
 class IWorld
 {
 public:
+    std::list<DelayedDamage> _delayedDamages;
+
     virtual ~IWorld() = default;
     [[nodiscard]] virtual WorldSession* FindSession(uint32 id) const = 0;
     [[nodiscard]] virtual WorldSession* FindOfflineSession(uint32 id) const = 0;
     [[nodiscard]] virtual WorldSession* FindOfflineSessionForCharacterGUID(ObjectGuid::LowType guidLow) const = 0;
+    virtual void AddDelayedDamage(Unit* attacker, Unit* victim, uint32 damage, CleanDamage const* cleanDamage, DamageEffectType damagetype, SpellSchoolMask damageSchoolMask, SpellInfo const* spellProto, bool durabilityLoss);
```

Comment:
why do we need this in IWorld?

Reply:
Or why in world at all? Why not just keep it in unit class?

Reply:
This is for the player and handled thru batch processing of the world.

Reply:
> why do we need this in IWorld?

We also had workflow and local ide scream about it needing to be present in both iworld and world.h you are welcome to see if u can find a solution.

Reply:
Why can't it be handled in Player::Update instead of World::Update?

Reply:
Would get errors with the player update.

Reply:
It just feels off

Reply:
I am open to assistance in consolidating.